File: Problems/0011.Container-With-Most-Water/ContainerWithMostWater.py
from typing import List

class Solution:

    # Runtime: 112 ms, faster than 99.74% of Python3 online submissions for Container With Most Water.
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        max_area = 0
        while l < r:
            if height[l] < height[r]:
                max_area = max(max_area, height[l] * (r-l))
                l += 1
            else:
                max_area = max(max_area, height[r] * (r-l))
                r -= 1
        return max_area

File: Problems/0011.Container-With-Most-Water/ContainerWithMostWater.go
package leetcode

func max(a, b int) int {
	if a <= b {
		return b
	}
	return a
}

// Runtime: 12 ms, faster than 92.60% of Go online submissions for Container With Most Water.
func maxArea(height []int) int {
	l := 0
	r := len(height) - 1
	maxArea := 0
	for l < r {
		if height[l] < height[r] {
			maxArea = max(maxArea, height[l]*(r-l))
			l++
		} else {
			maxArea = max(maxArea, height[r]*(r-l))
			r--
		}
	}
	return maxArea
}
